spec/date_spec.rb

```diff
@@ -0,0 +1,81 @@
+require_relative 'spec_helper'
+
+describe "DateRange Class" do
+  before do
+    @start_date = "2018-03-09"
+    @end_date = "2018-03-15"
+    @range = Hotel::DateRange.new(@start_date, @end_date)
+  end
+
+  describe "initializer" do
+    it "is an instance of DateRange" do
+      expect(@range).must_be_kind_of Hotel::DateRange
+    end
+
+    it "raises an error if start time is after end time" do
+      expect {
+        start_date = "2018-09-12"
+        end_date = "2018-09-03"
+        Hotel::DateRange.new(start_date, end_date)
+      }.must_raise ArgumentError
+    end
+  end
+  describe "Gets date range" do
+    it "inputted dates are converted to date range" do
+      expect(@range.get_date_range).must_be_kind_of Range
+    end
+  end
+
+  describe "checks for possible overlapping cases" do
+    it "checking overlap method for date that overlap" do
+      requested_start = "2018-03-09"
+      requested_end = "2018-03-15"
```

Comment:
I would like these tests to go into much more detail about what does and does not count as an overlap. A list of test cases was posted in slack, go check it out.